src/test/java/me/karakelley/http/RequestTest.java

```diff
@@ -97,6 +96,17 @@ void something() throws Exception {
     }
   }
 
+  @Test
+  void testNullPointerExceptionIsNotThrown() {
```

Comment:
I'd rather see this test define what _does_ happen in this situation, rather than just calling out one thing that doesn't happen, for a few reasons:
1. This high-level situation (a socket connection opening up but no data coming through) seems plausible (in fact, we've seen it happen when poking around the server already), and so this test could better serve as documentation of expected behavior. As-is, I don't know what should happen; I only know one thing that won't.
2. There are other app-crashing things that can happen beyond exceptions (see Error and Throwable), so I feel like this test doesn't sufficiently prove that your server won't blow up when this happens.